base/logging.h

```diff
@@ -82,16 +82,27 @@ void OutputDebugStringUTF8(const char *p);
 
 #ifdef _WIN32
 
+#define _LOG(s, ...) { \
+	const int BUFSIZE = 512; \
+	char buf[BUFSIZE + 2]; \
+	int len; \
+	len = snprintf(buf, BUFSIZE, "%s%s:%i: ", s, __FILE__, __LINE__); \
+	len += snprintf(buf + len, BUFSIZE - len, s __VA_ARGS__); \
+	sprintf(buf + len, "\n"); \
```

Comment:
So, this is actually not safe, since `snprintf()` is actually evil.

`snprintf()` returns the number of characters that **would** have been printed if the buffer was sufficiently large.  This is one of those "we chose to make the easy way the insecure way just because we're evil like that" functions (although it's useful if you're dynamically sizing the buffer, I suppose.)

So, if the extra `snprintf()`s should be protected by `if (len < BUFSIZE)` or etc.

-[Unknown]